### PR TITLE
MINOR: Clearer field names for ProducerIdsRecord and related classes

### PR DESCRIPTION
**Pull request by hachikuji**
I found the current naming of the fields a little confusing in regard to whether the block range was inclusive or exclusive. This patch tries to improve naming to make this clearer.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: metadata/src/main/java/org/apache/kafka/image/ProducerIdsDelta.java [hunk 1]
```diff
@@ -27,11 +27,11 @@ public ProducerIdsDelta(ProducerIdsImage image) {
         this.highestSeenProducerId = image.highestSeenProducerId();
     }
 
-    public void setHighestSeenProducerId(long highestSeenProducerId) {
+    public void setNextProducerId(long highestSeenProducerId) {
         this.highestSeenProducerId = highestSeenProducerId;
```
Comment (dengziming) [inline metadata/src/main/java/org/apache/kafka/image/ProducerIdsDelta.java]: maybe we should also change this field name to nextProducerId?

File: metadata/src/main/java/org/apache/kafka/image/ProducerIdsImage.java [hunk 2]
```diff
@@ -50,7 +50,7 @@ public void write(Consumer<List<ApiMessageAndVersion>> out) {
                 new ProducerIdsRecord().
                     setBrokerId(-1).
                     setBrokerEpoch(-1).
-                    setProducerIdsEnd(highestSeenProducerId), (short) 0)));
+                    setNextProducerId(highestSeenProducerId), (short) 0)));
```
Comment (dengziming) [inline metadata/src/main/java/org/apache/kafka/image/ProducerIdsImage.java]: ditto

File: metadata/src/main/java/org/apache/kafka/controller/ProducerIdControlManager.java [hunk 3]
```diff
@@ -33,49 +33,50 @@
 public class ProducerIdControlManager {
 
     private final ClusterControlManager clusterControlManager;
-    private final TimelineLong lastProducerId;
+    private final TimelineLong nextProducerId;
```
Comment (mumrah) [inline metadata/src/main/java/org/apache/kafka/controller/ProducerIdControlManager.java]: Probably worth adding a comment that this (TimelineLong) initializes to zero

**Review by mumrah — APPROVED** [review]
Thanks @hachikuji, this looks much better. We have some integration tests that check for contiguous PID blocks, so if those pass we should be good.

LGTM

**Pull request merged**